source/solvers/vof.cc

```diff
@@ -409,6 +409,105 @@ VolumeOfFluid<dim>::calculate_volume_and_mass(
     Utilities::MPI::sum(this->mass_monitored, mpi_communicator);
 }
 
+template <int dim>
+template <typename VectorType>
+void
+VolumeOfFluid<dim>::calculate_pressure_monitored_avg(
+  const TrilinosWrappers::MPI::Vector &solution,
+  const VectorType &                   current_solution_fd,
+  const Parameters::FluidIndicator     monitored_fluid)
+{
+  QGauss<dim>    quadrature_formula(this->cell_quadrature->size());
+  const MPI_Comm mpi_communicator = this->triangulation->get_communicator();
+
+  FEValues<dim> fe_values_vof(*this->mapping,
+                              *this->fe,
+                              *this->cell_quadrature,
+                              update_values);
+
+  const DoFHandler<dim> *dof_handler_fd =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+
+  FEValues<dim> fe_values_fd(*this->mapping,
+                             dof_handler_fd->get_fe(),
+                             quadrature_formula,
+                             update_values);
+
+  const unsigned int  n_q_points = quadrature_formula.size();
+  std::vector<double> local_phase_values(n_q_points);
+
+  const FEValuesExtractors::Scalar pressure(dim);
+  std::vector<double>              local_pressure_values(n_q_points);
+
+  this->pressure_monitored_avg = 0.;
+  int nb_values(0);
+
+  for (const auto &cell_vof : this->dof_handler.active_cell_iterators())
+    {
+      if (cell_vof->is_locally_owned())
+        {
+          fe_values_vof.reinit(cell_vof);
+          fe_values_vof.get_function_values(solution, local_phase_values);
+
+          // Get fluid dynamics active cell iterator
+          typename DoFHandler<dim>::active_cell_iterator cell_fd(
+            &(*(this->triangulation)),
+            cell_vof->level(),
+            cell_vof->index(),
+            dof_handler_fd);
+
+          fe_values_fd.reinit(cell_fd);
+          fe_values_fd[pressure].get_function_values(current_solution_fd,
+                                                     local_pressure_values);
+
+          for (unsigned int q = 0; q < n_q_points; q++)
+            {
+              // Gather minimum and maximum pressure on the phase of interest
+              if ((monitored_fluid == Parameters::FluidIndicator::fluid0 &&
+                   local_phase_values[q] < 0.5) ||
+                  (monitored_fluid == Parameters::FluidIndicator::fluid1 &&
+                   local_phase_values[q] > 0.5))
+                {
+                  this->pressure_monitored_avg += local_pressure_values[q];
```

Comment:
I saw you have defined this parameter in the header file, but I didn't see where you initialize it.

Reply:
It is in the same function, just before the loop on `cell_vof`. Do you think it should be elsewhere?

Reply:
well I would give it an initial value in the constructor of the class just to be sure, but still that's ok.
This brings the question though. Does this value really needs to be a member of the class vof or could it not just be a return argument of the function? It's best not to store things as class members when it can be avoided. If you just use it somewhere else, better return it and not alter the state.